html/browsers/browsing-the-web/scroll-to-fragid/fragment-and-encoding-2.html

```diff
@@ -0,0 +1,28 @@
+<!doctype html>
+<meta charset=windows-1252>
+<title>Fragment navigation: encoding</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<div id="log"></div>
+<div style=height:10000px></div>
+<div id=&#xFFFD;></div>
+<script>
+test(() => {
+  assert_equals(location.hash, "", "Page must be loaded with no hash");
+
+  location.hash = "%C2";
+  assert_equals(location.hash, "#%C2");
+  assert_greater_than(self.scrollY, 1000, "#%C2"); // UTF-8 decoded this'll match U+FFFD
+
```

Comment:
IMO, this test should be split into two or three.

test(() => {
 .... %C2
 ... %EF%BB%BF%C2
}, 'Invalid byte for UTF-8 should produce U+FFFD');

test(() => {
  .. .%EF%BF%BD
}, 'Percent-deocde result should be decoded as UTF-8 even if document encoding is windows-1252');

}, '